packages/theme/src/UIHelper/makeStyles.ts

```diff
@@ -22,7 +22,7 @@ export const { makeStyles } = createMakeStyles({ useTheme }) as any as {
         },
     ) => {
         classes: StyleOverrides extends { classes?: { [key in infer ExtraKeys]?: string | undefined } }
-            ? Record<ExtraKeys | RuleName, string>
+            ? Record<string extends ExtraKeys ? RuleName : ExtraKeys | RuleName, string>
```

Comment:
can you explain this change? It looks like `string extends ExtraKeys` always be true, therefore it will always go to the truthy branch.

Reply:
> It looks like string extends ExtraKeys always be true

If `StyleOverrides` is `{}`, then `ExtraKeys` would not be a union but `string`, and `string extends ExtraKeys` would be `false`

<img width="827" alt="image" src="https://user-images.githubusercontent.com/1141198/221343256-955b59f0-869b-4dd2-82ff-17941c4ac3fd.png">


https://www.typescriptlang.org/play?ssl=12&ssc=1&pln=13&pc=1#code/C4TwDgpgBAygjFAvFA3lAxgGwIYGdcS4BcqUA1iQOQBulUAvgwFCiSwBMSq9TL40AaQTJ4UCAA9gEAHYATXKSx4CuAPwk0AbTIQQUAJbSD0gGYQATlACik89gG7cAXXVRcwc4YDmUAD5QAVzkIE0MIWQZI1Wtbe0coEmkIagsmKCgAegyoAD1VPjYBThFOCSk5BTQlfEJXLR09Q2MzSxsPOJBnV3dPaR9-INkQsIjGRmi2uwdOhKgklPM0zOy8gugAJS4ivyhKatw4SiWs3PzeLIvLq+ub27vbtdgPABV+Lh7vR4BVaX1XtmQlAAEhBMJgAPZHR4wUQiF5vMoyeRuDzeKDRSggQh0KjSSHHFb5VjQGHFFG9HyIipPcz-aAYrG4HG7PFHdInVbfWFQH5-BGSJEKD59dG7RnMyisgmnLlk4WUgXU3l00WY7GzSX49mEoA

Reply:
> It looks like string extends ExtraKeys always be true

`string extends 'union' | 'type'`  would be `false`

Reply:
@Jack-Works Any other questions?